ddl/ddl_api.go

```diff
@@ -4241,7 +4241,9 @@ func (d *ddl) CreatePrimaryKey(ctx sessionctx.Context, ti ast.Ident, indexName m
 	}
 
 	indexName = model.NewCIStr(mysql.PrimaryKeyName)
-	if indexInfo := t.Meta().FindIndexByName(indexName.L); indexInfo != nil {
+	if indexInfo := t.Meta().FindIndexByName(indexName.L); indexInfo != nil ||
+		// If the table's PKIsHandle is true, it also means that this table has a primary key.
+		t.Meta().PKIsHandle {
```

Comment:
If we support the clustered index, I think need to add a check for `IsCommonHandle`. I don't do it, because other places also don't handle this problem.

Reply:
We had better extract these usages into a separate function.